sdk/samples/iot/pnp/pnp_thermostat_component.h

```diff
@@ -17,47 +17,118 @@
-    pnp_mqtt_message* mqtt_message);
+    az_span payload,
+    az_span* out_payload);
 
-az_result pnp_thermostat_process_property_update(
-    const az_iot_hub_client* client,
+/**
+ * @brief Build the thermostat's maximum temperature reported property message.
+ *
+ * @param[in] thermostat_component A pointer to the themostat component to get data.
+ * @param[in] payload An #az_span with sufficient capacity to hold the json payload.
+ * @param[out] out_payload A pointer to the #az_span containing the output json payload.
+ * @param[out] out_property_name The name of the reported property to be sent.
+ */
+void pnp_thermostat_build_maximum_temperature_reported_property(
     pnp_thermostat_component* thermostat_component,
+    az_span payload,
+    az_span* out_payload,
+    az_span* out_property_name);
+
+/**
+ * @brief Build the thermostat's error message with status.
+ *
+ * @param[in] component_name The name of the component for the reported property.
+ * @param[in] property_name The name of the property for which to send an update.
+ * @param[in] property_value The property value to be appended.
+ * @param[in] status The return status for the error message ack.
+ * @param[in] version The version for the reported property ack.
+ * @param[in] payload An #az_span with sufficient capacity to hold the json payload.
+ * @param[out] out_payload A pointer to the #az_span containing the output json payload.
+ */
+void pnp_thermostat_build_error_reported_property_with_status(
     az_span component_name,
-    const az_json_token* property_name,
-    const az_json_reader* property_value,
+    az_span property_name,
+    az_json_reader* const property_value,
+    az_iot_status status,
     int32_t version,
-    pnp_mqtt_message* mqtt_message);
+    az_span payload,
+    az_span* out_payload);
 
-az_result pnp_thermostat_process_command(
-    const az_iot_hub_client* client,
-    const pnp_thermostat_component* thermostat_component,
-    const az_iot_hub_client_method_request* command_request,
-    az_span component_name,
+/**
+ * @brief Update the thermostat's member variables and prepare reported property message.
+ *
+ * @param[in,out] ref_thermostat_component A pointer to the themostat component to update data.
+ * @param[in] property_name The name of the property to be updated.
+ * @param[in] property_value The value used for the property update.
+ * @param[in] version The version parsed from the received message, and used to prepare the returned
+ * reported property message.
+ * @param[in] payload An #az_span with sufficient capacity to hold the prepared reported property
+ * json payload.
+ * @param[out] out_payload A pointer to the #az_span containing the output reported property json
+ * payload.
+ */
+az_result pnp_thermostat_process_property_update(
+    pnp_thermostat_component* ref_thermostat_component,
+    az_json_token const* property_name,
+    az_json_reader const* const property_value,
```

Comment:
I don't think this is right. If we pass the `property_value` as an `az_json_reader`, this API will iterate on the reader which will change the thing it is pointed to. I think this should just be `az_json_reader property_value`

Reply:
i have changed back.  I did this change while I waiting to hear back from you on an earlier comment.